modules/core/src/test/java/org/apache/ignite/internal/processors/cache/IgniteOutOfMemoryPropagationTest.java

```diff
@@ -239,8 +260,86 @@ public void forceOOM(boolean useStreamer, TransactionConcurrency concurrency, Tr
         baseCfg.setBackups(this.backupsCnt);
         baseCfg.setWriteSynchronizationMode(this.writeSyncMode);
 
+        cfg.setMetricsLogFrequency(1_000);
+        cfg.setFailureHandler(new StopNodeOrHaltFailureHandler(false, 0));
         cfg.setCacheConfiguration(baseCfg);
 
         return cfg;
     }
+
+    /**
+     * Tests how eviction behaves when {@link DataRegionConfiguration#getEmptyPagesPoolSize()} is not enough for large
+     * values.
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testIgniteOOMWithoutNodeFailure() throws Exception {
+        atomicityMode = CacheAtomicityMode.ATOMIC;
+        mode = CacheMode.PARTITIONED;
+        writeSyncMode = CacheWriteSynchronizationMode.PRIMARY_SYNC;
+        backupsCnt = 0;
+        maxDataRegionSize = 1024L * 1024 * 1024;
+
+        IgniteEx sn = startGrids(1);
+
+        awaitPartitionMapExchange();
+
+        IgniteCache<Object, Object> cache = sn.cache(DEFAULT_CACHE_NAME);
+
+        List<IgniteInternalFuture<Object>> futures = new ArrayList<>();
+
+        Function<ThreadLocalRandom, byte[]> function = random -> new byte[random.nextInt(3) * 100 * 1024];
+
+        AtomicInteger exceptionsCnt = new AtomicInteger();
+
+        AtomicInteger successfulOpCountAfterException = new AtomicInteger();
+
+        final int exceptionsLimit = 2;
+
+        IgniteInClosure<IgniteInClosure<ThreadLocalRandom>> task = op -> {
+            ThreadLocalRandom random = ThreadLocalRandom.current();
+
+            while (true) {
+                if (exceptionsCnt.get() >= exceptionsLimit)
+                    break;
+
+                try {
+                    op.apply(random);
+
+                    if (exceptionsCnt.get() > 0)
```

Comment:
I do not understand the conditions, I think you need without it.

Reply:
Discussed this in slack, stays as is.